R/check_enough_data.R

```diff
@@ -0,0 +1,201 @@
+#' Check the dataset contains enough data points.
+#'
+#' `check_enough_data` creates a *specification* of a recipe
+#'  operation that will check if variables contain enough data.
+#'
+#' @param recipe A recipe object. The check will be added to the
+#'  sequence of operations for this recipe.
+#' @param ... One or more selector functions to choose variables for this check.
+#'  See [selections()] for more details. You will usually want to use
+#'  [recipes::all_predictors()] and/or [recipes::all_outcomes()] here.
+#' @param n The minimum number of data points required for training. If this is
+#'   NULL, the total number of predictors will be used.
+#' @param epi_keys A character vector of column names on which to group the data
+#'   and check threshold within each group. Useful if your forecaster trains
+#'   per group (for example, per geo_value).
+#' @param drop_na A logical for whether to count NA values as valid rows.
+#' @param role Not used by this check since no new variables are
+#'  created.
+#' @param trained A logical for whether the selectors in `...`
+#' have been resolved by [prep()].
+#' @param columns An internal argument that tracks which columns are evaluated
+#'   for this check. Should not be used by the user.
```

Comment:
Let's remove this arg here. It can be added / used in the `*_new()` constructor.

(Core `{recipes}` exposes args that the user shouldn't and wouldn't want to change. My preference is not to have them unless explicitly mentioned as important (`role` would be an example here)).